climada/hazard/storm_europe.py

```diff
@@ -199,6 +206,209 @@ def _read_one_nc(self, file_name, centroids):
         ncdf.close()
         return new_haz
 
+
+    def read_cosmoe_file(self, fp_file, run_date, event_date=None,
+                         model_name='COSMO-2E', description=None):
+        """Clear instance and read MeteoSwiss cosmoe weather forecast
+        footprint netcdf files into it. One event is one full day in UCT.
+        Works for COSMO-1E and COSMO-2E
+
+        Parameters:
+            fp_file (str): string directing to one netcdf file
+            run_date (datetime): The starting timepoint of the forecast run
+                of the cosmo model
+            event_date (datetime, optional): one day within the forecast
+                period, only this day (00H-24H) will be included in the hazard
+            model_name (str,optional): select the name of the icon model to
+                be downloaded. Must match the url string
+                (see _download_icon_grib for further info)
+            description (str, optional): description of the events, defaults
+                to 'cosmo weather forecast'
+        """
+        self.clear()
+        # create centroids
+        self.centroids = self._centroids_from_nc(fp_file)
+
+        # read intensity from file
+        ncdf = xr.open_dataset(fp_file)
+        ncdf = ncdf.assign_coords(date=('time',ncdf["time"].dt.floor("D")))
+
+        if event_date:
```

Comment:
it could be worth adding a check that the `event_date` is covered by the input file plus an informative error. right now when it fails the code gives ~100 lines of error message.

Reply:
I implemented this, good suggestion.